torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -765,68 +781,58 @@ def _get_ignored_modules(
             if not isinstance(child, FullyShardedDataParallel) and
             not isinstance(child, FlattenParamsWrapper)
         )
+        if root_module in ignored_modules:
+            raise ValueError(
+                "Trying to ignore the top-level module passed into the FSDP "
+                "constructor itself will result in all parameters being "
+                f"ignored and is not supported: {module}"
+            )
         return ignored_modules
 
-    def _get_ignored_parameters(self) -> Set[torch.nn.Parameter]:
+    def _get_ignored_parameters(
+        self,
+        ignored_modules: Set[torch.nn.Module],
+    ) -> Set[torch.nn.Parameter]:
         """Returns the parameters of the modules in ``ignored_modules`` as a
         :class:`set`, excluding any :class:`FlatParameter` s.
         """
-        assert hasattr(self, "_ignored_modules"), \
-            "Expects `self._ignored_modules` to be initialized"
         return set(
-            p for m in self._ignored_modules for p in m.parameters()
+            p for m in ignored_modules for p in m.parameters()
             if not isinstance(p, FlatParameter)
         )
 
-    def _get_ignored_named_tensors(
+    def _get_ignored_buffers(
```

Comment:
Do these need to be separate functions or can they just work together to return buffers and params since there seems to only be one use case?

Reply:
This can be combined with below, but since the next diff overrides this anyway, I will leave it as is.